### PR TITLE
feat: add possibility to manage service account

### PR DESCRIPTION
**Pull request by mirkoszy**
## Related issue(s)
https://github.com/ory/k8s/issues/348

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Comment by Demonsthere** [comment]
Hello there,
The pipeline is failing because of an upstream issue we found in `oryd/kratos-selfservice-ui-node:latest`, which is still used in your branch. Please pull the newest master into your branch so the CI can pass! :)

File: helm/charts/hydra/values.yaml [hunk 1]
```diff
@@ -217,7 +217,13 @@ deployment:
   # In some situations it is needed to provides specific permissions to Hydra deployments
   # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
   # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.
-  serviceAccountName: ""
+  serviceAccount:
+    # -- Specifies whether a service account should be created
+    create: true
+    # Annotations to add to the service account
```
Comment (Demonsthere) [inline helm/charts/hydra/values.yaml:223]:
```suggestion
    # -- Annotations to add to the service account
```

File: helm/charts/kratos/values.yaml [hunk 2]
```diff
@@ -258,6 +258,14 @@ deployment:
   # https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables
   # environmentSecretsName:
 
+  serviceAccount:
+    # -- Specifies whether a service account should be created
+    create: true
+    # Annotations to add to the service account
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:264]:
```suggestion
    # -- Annotations to add to the service account
```

File: helm/charts/oathkeeper/values.yaml [hunk 3]
```diff
@@ -176,6 +176,14 @@ deployment:
     allowPrivilegeEscalation: false
     privileged: false
 
+  serviceAccount:
+    # -- Specifies whether a service account should be created
+    create: true
+    # Annotations to add to the service account
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/values.yaml:182]:
```suggestion
    # -- Annotations to add to the service account
```

**Review by Demonsthere — COMMENTED** [review]
Small housekeeping. I will do some manual tests, in the meantime please sync with master to unblock CI and we can merge it right away ;)

**Comment by mirkoszy** [comment]
I synced to the master. There are more failing jobs now not associated with my changes.

File: helm/charts/oathkeeper/values.yaml [hunk 4]
```diff
@@ -186,6 +186,14 @@ deployment:
     allowPrivilegeEscalation: false
     privileged: false
 
+  serviceAccount:
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/values.yaml:189]:
```suggestion
  # -- Specify the serviceAccountName value.
  # In some situations it is needed to provides specific permissions to Hydra deployments
  # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
  # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.**
  serviceAccount:
```

File: helm/charts/kratos/values.yaml [hunk 5]
```diff
@@ -265,6 +265,14 @@ deployment:
   # https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#configure-all-key-value-pairs-in-a-secret-as-container-environment-variables
   # environmentSecretsName:
 
+  serviceAccount:
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:268]:
```suggestion
  # -- Specify the serviceAccountName value.
  # In some situations it is needed to provides specific permissions to Hydra deployments
  # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
  # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.
  serviceAccount:
```

File: helm/charts/hydra/values.yaml [hunk 6]
```diff
@@ -230,7 +230,13 @@ deployment:
   # In some situations it is needed to provides specific permissions to Hydra deployments
   # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
   # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.
-  serviceAccountName: ""
+  serviceAccount:
```
Comment (Demonsthere) [inline helm/charts/hydra/values.yaml:233]:
```suggestion
  # -- Specify the serviceAccountName value.
  # In some situations it is needed to provides specific permissions to Hydra deployments
  # Like for example installing Hydra on a cluster with a PosSecurityPolicy and Istio.
  # Uncoment if it is needed to provide a ServiceAccount for the Hydra deployment.
  serviceAccount:
```

**Review by Demonsthere — APPROVED** [review]
Just patching up the comments

**Pull request merged**